duplex_http_call.go

```diff
@@ -130,179 +116,196 @@ func (d *duplexHTTPCall) CloseWrite() error {
 	// forever. To make sure users don't have to worry about this, the generated
 	// code for unary, client streaming, and server streaming RPCs must call
 	// CloseWrite automatically rather than requiring the user to do it.
-	return d.requestBodyWriter.Close()
+	if c.requestBodyWriter != nil {
+		return c.requestBodyWriter.Close()
```

Comment:
Ensure we keep the close semantics in https://github.com/connectrpc/connect-go/pull/594